pkg/virt-controller/watch/network.go

```diff
@@ -91,19 +93,37 @@ func trimDoneInterfaceRequests(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInst
 	vm.Status.InterfaceRequests = updateIfaceRequests
 }
 
-func handleDynamicInterfaceRequests(vm *v1.VirtualMachine) {
+func handleDynamicInterfaceRequests(vm *v1.VirtualMachine, vmiHasPodNetwork bool, clusterConfig *virtconfig.ClusterConfig) error {
 	if len(vm.Status.InterfaceRequests) == 0 {
-		return
+		return nil
 	}
 
 	vmTemplateCopy := vm.Spec.Template.Spec.DeepCopy()
+
+	// Explicitly add the default interface before hot-plugging another interface
+	if vmiHasPodNetwork && len(vmTemplateCopy.Domain.Devices.Interfaces) == 0 {
```

Comment:
The boolean argument to this function and this code block is a hint that it should not be embedded here, but defined as a separated function and called by the caller explicitly when needed.